internal/pool/pool.go

```diff
@@ -302,9 +302,8 @@ func (p *ConnPool) popIdle() *Conn {
 		return nil
 	}
 
-	idx := len(p.idleConns) - 1
-	cn := p.idleConns[idx]
-	p.idleConns = p.idleConns[:idx]
+	cn := p.idleConns[0]
+	p.idleConns = p.idleConns[1:]
```

Comment:
As I understand there is a memory leak here since first element of the slice is inaccessible but can not be freed.

Reply:
maybe below code could fix the memory leak ? @vmihailenco 
`
cn := p.idleConns[0]
p.idleConns[0] = nil
p.idleConns = p.idleConns[1:]
`

Reply:
What is the motivation? Why balancing connection usage is important?

Reply:
Because unbalanced connection usage makes few backends bear more traffics that will increase the average response time per query.

Reply:
Do you mean postgres backends or you are using some load balancer that is not able to re-distribute queries that come on the same connection?

Reply:
I mean postgres backends.

Reply:
Okay, thanks. Before making any changes we need a benchmark or something that proves that uneven queries distribution between Postgres backends affects performance.

Reply:
Actually we happen to encounter this issue in our production. I can provide some screenshots from grafana graphs showing how the response time affected by uneven queries. But it's non-trivial to give a minimal demo of code to reproduce this issue.